src/halmos/__main__.py

```diff
@@ -1195,8 +1224,19 @@ def main() -> int:
                     total_passed += num_passed
                     total_failed += num_failed
 
+                    if test_results_map is not None:
+                        if contract_path in test_results_map:
+                            raise ValueError("already exists", contract_path)
+                        test_results_map[contract_path] = list(
```

Comment:
arbitrary nit, I prefer the comprehension version:

`test_results_map[contract_path] = [asdict(t) for t in test_results]`